### ISSUE TITLE
[AQ40] Twin Emperors - Teleport 

### ISSUE THREAD

**Issue by BeefSteaker** [azerothcore/azerothcore-wotlk#13523]
https://github.com/chromiecraft/chromiecraft/issues/4264
### What client do you play on?

enUS

### Faction

Both

### Content Phase:

60

### Current Behaviour

Vek’lor and Vek’nilash start their attacks right after swapping, sometimes this makes the two bosses go after their previous targets.
Vek’nilash for example may attempt to go after its previous target to try to channel a new shadowbolt.
https://www.twitch.tv/videos/1629467401?t=03h00m06s

### Expected Blizzlike Behaviour

Vek’lor and Vek’nilash should have a delay of 2 seconds before retaking their attacks after the swamp.
They complete full teleport animation, remove anyone on target, stop all actions and then retake their normal cycle of attacks.

### Source

explanation: https://youtu.be/ulzbebfPxcg?t=261

example 1: https://youtu.be/ulzbebfPxcg?t=194
example 2: https://youtu.be/ulzbebfPxcg?t=233
example 3: https://youtu.be/ulzbebfPxcg?t=298
example 4: https://youtu.be/ulzbebfPxcg?t=330

another run: https://youtu.be/4CmadeEllDk?t=68

### Steps to reproduce the problem

Go c id 15276
Engage combat in this encounter

### Extra Notes

Without this 'grace period' , the fight gets buggy with threat and more challenging than it was designed to be as that short delay allows for the other tank/warlock to react and take initial action.

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/b427e8e18cb6baa80124ce8e3a7e79aa670bcc74

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-bg-item-reward](https://github.com/azerothcore/mod-bg-item-reward)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-cta-switch](https://github.com/azerothcore/mod-cta-switch)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-dmf-switch](https://github.com/azerothcore/mod-dmf-switch)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna](https://github.com/azerothcore/mod-eluna)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-low-level-rbg](https://github.com/azerothcore/mod-low-level-rbg)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-progression-system](https://github.com/azerothcore/mod-progression-system)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-rdf-expansion](https://github.com/azerothcore/mod-rdf-expansion)
- [mod-transmog](https://github.com/azerothcore/mod-transmog)
- [mod-weekend-xp](https://github.com/azerothcore/mod-weekend-xp)
- [mod-instanced-worldbosses](https://github.com/nyeriah/mod-instanced-worldbosses)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

**Comment by Gultask** [comment]
Have you found a way to reproduce it outside a raid setting? Because it always worked for me.

**Comment by BeefSteaker** [comment]
No, it's just been consistent in the attempts I've been watching

**Comment by Gultask** [comment]
@BeefSteaker Could you mention that in the Steps to Reproduce, then? The mechanic is already implemented but something's breaking it and we don't know what.

**Comment by Si1ker** [comment]
> 

@Gultask With "mechanic is already implemented" are you also taking in consideration the main point of the report: the 2 sec delay on both bosses after each TP?

A lot of proof that it should be there, ty.

**Comment by Gultask** [comment]
It is there but there's something messing it up.

If you try it using 2 characters it'll probably be perfect.

**Comment by Si1ker** [comment]
https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_twinemperors.cpp#L164
huh, ye def not taking place in the encounter, more than +50 attempts and it was instant every time (I was the warlock tanking the first kill on Twins in ChromieCraft)
